pilot/cmd/pilot-agent/main.go

```diff
@@ -534,6 +535,9 @@ func init() {
 	proxyCmd.PersistentFlags().StringVar(&proxyLogLevel, "proxyLogLevel", "warning",
 		fmt.Sprintf("The log level used to start the Envoy proxy (choose from {%s, %s, %s, %s, %s, %s, %s})",
 			"trace", "debug", "info", "warning", "error", "critical", "off"))
+	// See https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-component-log-level
+	proxyCmd.PersistentFlags().StringVar(&proxyComponentLogLevel, "proxyComponentLogLevel", "misc:error",
```

Comment:
@mandarjog are we concerned with new flags and backward-compat? Should we prefer environment variables?

If the answer is yes, it might be appropriate to replace all of our flags with environment variables (deprecate all existing flags), so that people don't go down the wrong path in the future.

Reply:
@mandarjog is this an issue?